types/field_type.go

```diff
@@ -365,8 +365,8 @@ func DefaultTypeForValue(value interface{}, tp *FieldType) {
 		// TODO: tp.Flen should be len(x) * 3 (max bytes length of CharsetUTF8)
 		tp.Flen = len(x)
 		tp.Decimal = UnspecifiedLength
-		tp.Charset = mysql.DefaultCharset
-		tp.Collate = mysql.DefaultCollationName
+		tp.Charset = charset.CharsetUTF8MB4
```

Comment:
Can we use GetDefaultCharsetAndCollate?

Reply:
This change makes sense to me for this PR. My PR will change it back though :)